onboardingtaskapp.client/src/components/Customer/List.jsx

```diff
@@ -9,39 +9,78 @@ export default function List() {
     const [customers, setCustomers] = useState([]);       
 
     useEffect(() => {       
-        fetch('https://localhost:7207/api/Customer')
-            .then(response => response.json())
-            .then(data => setCustomers(data))
-            .catch(error => console.error('Error fetching customers:', error));
+        fetchCustomers();
     }, []);
 
+    const fetchCustomers = async () => {
 
-    const handleDelete = async (customerId) => {
         try {
-            const response = await fetch(`https://localhost:7207/api/Customer/${customerId}`, {
-                method: 'DELETE',
-                headers: {
-                    'Content-Type': 'application/json',
-                },
-            });
-            if (!response.ok) {
-                throw new Error('Network response was not ok');
-            }
-            // Handle success
-            window.location.replace("/");
-            console.log('Customer deleted successfully');
-            
+            await fetch('https://onboardsite.azurewebsites.net/api/Customer')
+                .then(response => response.json())
+                .then(data => setCustomers(data))
+                .catch(error => console.error('Error fetching customers:', error));
+
         } catch (error) {
-            console.error('There was a problem deleting customer:', error.message);
+            console.error('There was a problem fetching customers data:', error.message);
+        }       
+    }
+
+    const handleDeleteSuccess = async (isDeleted) => {
+
+        if (isDeleted) {
+           await fetchCustomers();
+        }
+    }
+
+    const handleUpdateSuccess = async (isUpdated) => {
+
+        if (isUpdated) {
+            await fetchCustomers();
         }
-    };
+    }
+
+    const handleCreateSuccess = async (isCreated) => {
+
+        if (isCreated) {
+            await fetchCustomers();
+        }
+    }
+
+    const renderTableRows = (customer, handleUpdateSuccess, handleDeleteSuccess) => {
+
+        const customerDeleteItem = {
+            id: customer.id,
+            title: 'Delete Customer',
+            buttonText: 'DELETE',
+            url: 'Customer',
+            redirect: ''
+        };
+
+        return (
+            <tr key={customer.id}>
+                <td data-label="Name">{customer.name}</td>
+                <td data-label="Address">{customer.address}</td>
+                <td>
+                    <CustomerEdit
+                        item={{ customer }}
+                        isUpdated={handleUpdateSuccess}
+                    />
+                </td>
+                <td>
+                    <CustomerDelete
+                        item={customerDeleteItem }                       
+                        isDeleted={handleDeleteSuccess}
 
+                    />
```

Comment:
This is fine, but I suggest you should consider using **() => handleUpdateSuccess()** in here. This is because if you'll get used in this format you might overlook the possibility that this kind of format in functions might trigger automatically when the page load initially. Please implement to other pages. Thanks